apps/remote_control/test/lexical/remote_control/code_intelligence/rename_test.exs

```diff
@@ -0,0 +1,338 @@
+            alias TopLevel.Foo, as: Parent
+            |Parent.Bar
+          end
+        ]
+        |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Foo.Bar do]
+
+      assert result =~ ~S[alias TopLevel.Foo, as: Renamed]
+      assert result =~ ~S[  Renamed.Bar]
+    end
+
+    test "shouldn't rename the relative module" do
+      {:ok, result} =
+        ~q[
+        defmodule |Foo do
+        end
+
+        defmodule FooTest do
+        end
+        ] |> rename("Renamed")
+
+      assert result =~ ~S[defmodule FooTest do]
+    end
+
+    test "shouldn't rename the descendants when the cursor at the end of the module" do
+      {:ok, result} = ~q[
+        defmodule TopLevel.Module do # ✓
+        end
+
+        defmodule TopLevel.Module.Another do # x
+          alias TopLevel.|Module
+        end
+      ] |> rename("Renamed")
+
+      refute result =~ ~S[defmodule TopLevel.Renamed.Another]
+
+      assert result =~ ~S[defmodule TopLevel.Renamed do]
+      assert result =~ ~S[alias TopLevel.Renamed]
+    end
+  end
+
+  describe "rename descendants" do
+    test "in the middle of definition" do
+      {:ok, result} =
+        ~q[
+          defmodule TopLevel.|Middle.Module do
+            alias TopLevel.Middle.Module
+          end
+        ] |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Renamed.Module]
+      assert result =~ ~S[alias TopLevel.Renamed.Module]
+    end
+
+    test "in the middle of reference" do
+      {:ok, result} =
+        ~q[
+          defmodule TopLevel.Second.Middle.Module do
+            alias TopLevel.Second.|Middle.Module
+          end
+        ] |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Second.Renamed.Module]
+      assert result =~ ~S[alias TopLevel.Second.Renamed.Module]
+    end
+
+    test "succeeds when there are same module name in the cursor neighborhood" do
+      {:ok, result} =
+        ~q[
+          defmodule TopLevel.Foo do
+          end
+
+          defmodule TopLevel.Foo.Foo do
+          end
+
+          defmodule TopLevel.Another do
+            alias TopLevel.Foo.|Foo
+          end
+        ] |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Foo do]
+      assert result =~ ~S[defmodule TopLevel.Foo.Renamed do]
+      assert result =~ ~S[alias TopLevel.Foo.Renamed]
+    end
+
+    test "succeeds even if there are descendants with the same name" do
+      {:ok, result} =
+        ~q[
+        defmodule TopLevel.Foo do
+          defmodule Foo do # skip this
+          end
+        end
+
+        defmodule TopLevel.Bar do
+          alias TopLevel.|Foo.Foo
+        end
+      ]
+        |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Renamed do]
+      assert result =~ ~S[defmodule Foo do # skip this]
+      assert result =~ ~S[alias TopLevel.Renamed.Foo]
+    end
+  end
+
+  describe "rename struct" do
+    test "succeeds when the cursor on the definition" do
```

Comment:
When we hover, before the first letter of the definition, it returns the `module`. So, it is reasonable here. In this case, I think using `"at"` instead of `"before"` would be more appropriate.